src/l1-resources/private-endpoint/index.ts

```diff
@@ -15,134 +15,75 @@ export interface CfnPrivateEndpointProps {
-   *
-   * @schema CfnPrivateEndpointProps#EndpointServiceName
-   */
-  readonly endpointServiceName?: string;
-
-  /**
-   * Error message pertaining to the AWS PrivateLink connection. Returns null if there are no errors.
-   *
-   * @schema CfnPrivateEndpointProps#ErrorMessage
-   */
-  readonly errorMessage?: string;
-
-  /**
-   * Status of the Atlas PrivateEndpoint service connection
-   *
-   * @schema CfnPrivateEndpointProps#Status
-   */
-  readonly status?: string;
-
   /**
    * Unique 24-hexadecimal digit string that identifies your project.
    *
-   * @schema CfnPrivateEndpointProps#GroupId
+   * @schema CfnPrivateEndpointProps#ProjectId
    */
-  readonly groupId: string;
+  readonly projectId?: string;
 
   /**
-   * Aws Region
+   * Unique 24-hexadecimal digit string that identifies the private endpoint service for which you want to create a private endpoint.
    *
-   * @schema CfnPrivateEndpointProps#Region
+   * @schema CfnPrivateEndpointProps#EndpointServiceId
    */
-  readonly region: string;
-
-  /**
-   * List of private endpoint associated to the service
-   *
-   * @schema CfnPrivateEndpointProps#PrivateEndpoints
-   */
-  readonly privateEndpoints?: PrivateEndpoint[];
-}
-
-/**
- * Converts an object of type 'CfnPrivateEndpointProps' to JSON representation.
- */
-/* eslint-disable max-len, quote-props */
-export function toJson_CfnPrivateEndpointProps(
-  obj: CfnPrivateEndpointProps | undefined
-): Record<string, any> | undefined {
-  if (obj === undefined) {
-    return undefined;
-  }
-  const result = {
-    Profile: obj.profile,
-    EndpointServiceName: obj.endpointServiceName,
-    ErrorMessage: obj.errorMessage,
-    Status: obj.status,
-    GroupId: obj.groupId,
-    Region: obj.region,
-    PrivateEndpoints: obj.privateEndpoints?.map((y) =>
-      toJson_PrivateEndpoint(y)
-    ),
-  };
-  // filter undefined values
-  return Object.entries(result).reduce(
-    (r, i) => (i[1] === undefined ? r : { ...r, [i[0]]: i[1] }),
-    {}
-  );
-}
-/* eslint-enable max-len, quote-props */
+  readonly endpointServiceId?: string;
 
-/**
- * @schema PrivateEndpoint
- */
-export interface PrivateEndpoint {
   /**
-   * String Representing the AWS VPC ID (like: vpc-xxxxxxxxxxxxxxxx) (Used For Creating the AWS VPC Endpoint)
+   * Unique 24-hexadecimal digit string that identifies the private endpoint service for which you want to create a private endpoint.
    *
-   * @schema PrivateEndpoint#VpcId
+   * @schema CfnPrivateEndpointProps#InterfaceEndpointId
    */
-  readonly vpcId?: string;
+  readonly interfaceEndpointId?: string;
 
   /**
-   * List of string representing the AWS VPC Subnet ID (like: subnet-xxxxxxxxxxxxxxxxx) (Used For Creating the AWS VPC Endpoint)
+   * Cloud service provider that manages this private endpoint.
    *
-   * @schema PrivateEndpoint#SubnetIds
+   * @schema CfnPrivateEndpointProps#CloudProvider
    */
-  readonly subnetIds?: string[];
+  readonly cloudProvider?: string;
```

Comment:
FYI: this is a breaking change which means that we need to increase the Major version of the cdk. I updated the title and description of the ticket to make this clear